keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlDeliveryConfigRepository.kt

```diff
@@ -29,43 +29,35 @@ class SqlDeliveryConfigRepository(
   private val mapper = configuredObjectMapper()
 
   override fun store(deliveryConfig: DeliveryConfig) {
-    jooq.inTransaction {
-      val uid = randomUID().toString()
-      with(deliveryConfig) {
-        insertInto(DELIVERY_CONFIG)
-          .set(DELIVERY_CONFIG.UID, uid)
-          .set(DELIVERY_CONFIG.NAME, name)
-          .set(DELIVERY_CONFIG.APPLICATION, application)
+    val uid = randomUID().toString()
+    with(deliveryConfig) {
+      jooq.insertInto(DELIVERY_CONFIG)
+        .set(DELIVERY_CONFIG.UID, uid)
+        .set(DELIVERY_CONFIG.NAME, name)
+        .set(DELIVERY_CONFIG.APPLICATION, application)
+        .onDuplicateKeyIgnore()
+        .execute()
+      artifacts.forEach { artifact ->
+        jooq.insertInto(DELIVERY_CONFIG_ARTIFACT)
+          .set(DELIVERY_CONFIG_ARTIFACT.DELIVERY_CONFIG_UID, uid)
+          .set(DELIVERY_CONFIG_ARTIFACT.ARTIFACT_UID, jooq.select(DELIVERY_ARTIFACT.UID).from(DELIVERY_ARTIFACT).where(DELIVERY_ARTIFACT.NAME.eq(artifact.name)))
           .onDuplicateKeyIgnore()
           .execute()
-        artifacts.forEach { artifact ->
-          insertInto(DELIVERY_ARTIFACT)
-            .set(DELIVERY_ARTIFACT.UID, randomUID().toString())
-            .set(DELIVERY_ARTIFACT.NAME, artifact.name)
-            .set(DELIVERY_ARTIFACT.TYPE, artifact.type.name)
-            .onDuplicateKeyIgnore()
-            .execute()
-          insertInto(DELIVERY_CONFIG_ARTIFACT)
-            .set(DELIVERY_CONFIG_ARTIFACT.DELIVERY_CONFIG_UID, uid)
-            .set(DELIVERY_CONFIG_ARTIFACT.ARTIFACT_UID, select(DELIVERY_ARTIFACT.UID).from(DELIVERY_ARTIFACT).where(DELIVERY_ARTIFACT.NAME.eq(artifact.name)))
-            .onDuplicateKeyIgnore()
-            .execute()
-        }
-        environments.forEach { environment ->
-          val environmentUID = randomUID().toString()
-          insertInto(ENVIRONMENT)
-            .set(ENVIRONMENT.UID, environmentUID)
-            .set(ENVIRONMENT.DELIVERY_CONFIG_UID, uid)
-            .set(ENVIRONMENT.NAME, environment.name)
+      }
+      environments.forEach { environment ->
+        val environmentUID = randomUID().toString()
+        jooq.insertInto(ENVIRONMENT)
+          .set(ENVIRONMENT.UID, environmentUID)
+          .set(ENVIRONMENT.DELIVERY_CONFIG_UID, uid)
+          .set(ENVIRONMENT.NAME, environment.name)
+          .onDuplicateKeyIgnore()
```

Comment:
Not sure about environments always getting a new uid here. This store method is called for delivery config upsert, so it seems like the environment table is likely to get lots of "duplicate" rows, where `delivery_config_uid` and `name` are the same, but with a new `uid` per config upsert.